dowhy/causal_estimators/instrumental_variable_estimator.py

```diff
@@ -139,17 +139,18 @@ def estimate_effect(
             instrument_is_binary = num_unique_values <= 2
             if instrument_is_binary:
                 # Obtain estimate by Wald Estimator
-                y1_z = np.mean(data[self._target_estimand.outcome_variable[0]][instrument == 1])
-                y0_z = np.mean(data[self._target_estimand.outcome_variable[0]][instrument == 0])
-                x1_z = np.mean(data[self._target_estimand.treatment_variable[0]][instrument == 1])
-                x0_z = np.mean(data[self._target_estimand.treatment_variable[0]][instrument == 0])
+                y1_z = np.nanmean(data[self._target_estimand.outcome_variable[0]][instrument == 1])
```

Comment:
I understand the motivation, but sometimes the user may not be aware of the missing data (especially when it may be due to a bug in their data processing code). 
For this reason, can we issue a warning message as this nanmean operation is done?

the code structure can be: try np.mean operations, if it fails, then we issue a warning that "The dataset contains missing data, proceeding to ignore it", and then using nanmean as a fallback. 
for warning, you can use, `self.logger.warning("message")`